linker/linker_soinfo.cpp

```diff
@@ -296,7 +297,13 @@ static bool symbol_matches_soaddr(const ElfW(Sym)* sym, ElfW(Addr) soaddr) {
 }
 
 ElfW(Sym)* soinfo::gnu_addr_lookup(const void* addr) {
-  ElfW(Addr) soaddr = reinterpret_cast<ElfW(Addr)>(addr) - load_bias;
+  ElfW(Addr) soaddr = reinterpret_cast<ElfW(Addr)>(addr);
+  SegmentInfo* seg_info = find_rand_segment(soaddr);
+  if (seg_info == nullptr) {
+    soaddr -= load_bias;
+  } else {
+    soaddr += seg_info->phdr_addr - seg_info->mem_addr;
+  }
```

Comment:
Still think this should be `{is|in}_rand_segment`.  KISS
`find_rand_segment` is never used to *find*, only always to *check*.

```cpp
if (is_rand_segment(soaddr)) {
  soaddr -= load_bias;
} else {
  soaddr += seg_info->phdr_addr - seg_info->mem_addr;
}
```